core/nodepool/config/templates/stack-template.json

```diff
@@ -91,7 +91,18 @@
                   {{end}}
                   "VolumeType": "{{$spec.RootVolumeType}}"
                 }
+              }{{range $volumeMountSpecIndex, $volumeMountSpec := $.SpotFleet.VolumeMounts}},
```

Comment:
How about changing it to `$.Worker.VolumeMounts` instead of `$.SpotFleet.VolumeMounts` and completely removing the `VolumeMounts` field inside the `SpotFleet` struct?
Doing so would reduce some code duplication.